universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -159,40 +164,93 @@ public Program createOrUpdateDraft(Program existingProgram) {
     return database
         .find(Program.class)
         .where()
-        .eq(
-            "name",
-            database.find(Program.class).setId(programId).select("name").findSingleAttribute())
+        .in("name", database.find(Program.class).select("name").where().eq("id", programId).query())
+        .query()
         .findList()
         .stream()
         .collect(ImmutableList.toImmutableList());
   }
 
-  public ImmutableList<Long> getAllProgramVersionIds(long programId) {
+  public PaginationResult<Application> getApplicationsForAllProgramVersions(
+      long programId, PaginationSpec paginationSpec, Optional<String> searchNameFragment) {
+    ExpressionList<Application> query =
+        database
+            .find(Application.class)
+            .fetch("program")
+            .orderBy("id desc")
+            .where()
+            .in("program_id", allProgramVersionsQuery(programId))
+            .in(
+                "lifecycle_stage",
+                ImmutableList.of(LifecycleStage.ACTIVE, LifecycleStage.OBSOLETE));
+
+    if (searchNameFragment.isPresent() && !searchNameFragment.get().isBlank()) {
+      String search = searchNameFragment.get();
+
+      if (search.matches("^\\d+$")) {
+        query = query.or().eq("id", search).eq("applicant_id", search).endOr();
+      } else {
+        search = search.toLowerCase(java.util.Locale.ROOT);
+
+        query =
+            query
+                .or()
+                .eq("submitter_email", search)
+                .raw(
+                    getApplicationObjectPath("applicant", "name", "first_name") + " ILIKE ?",
```

Comment:
The super fuzzy search here should be in the doc comment. email or first or last name is unintuitive.

Reply:
updated doc, lmk how it looks